sup3r/qa/stats.py

```diff
@@ -0,0 +1,494 @@
+               "t_elap = time.time() - t0;\n")
+
+        job_name = config.get('job_name', None)
+        if job_name is not None:
+            status_dir = config.get('status_dir', None)
+            status_file_arg_str = f'"{status_dir}", '
+            status_file_arg_str += f'module="{ModuleName.WIND_STATS}", '
+            status_file_arg_str += f'job_name="{job_name}", '
+            status_file_arg_str += 'attrs=job_attrs'
+
+            cmd += ('job_attrs = {};\n'.format(json.dumps(config)
+                                               .replace("null", "None")
+                                               .replace("false", "False")
+                                               .replace("true", "True")))
+            cmd += 'job_attrs.update({"job_status": "successful"});\n'
+            cmd += 'job_attrs.update({"time": t_elap});\n'
+            cmd += f'Status.make_job_file({status_file_arg_str})'
+
+        cmd += (";\'\n")
+
+        return cmd.replace('\\', '/')
+
+    @staticmethod
+    def get_wind_stats(u, v):
+        """Get stats for wind fields
+
+        Parameters
+        ----------
+        u: ndarray
+            Longitudinal velocity component
+            (lat, lon, temporal)
+        v : ndarray
+            Latitudinal velocity component
+            (lat, lon, temporal)
+
+        Returns
+        -------
+        stats : dict
+            Dictionary of stats for wind fields
+        """
+        v_dist = velocity_gradient_dist(u)
+        vort_dist = vorticity_dist(u, v)
+        rr_dist = ramp_rate_dist(u, v)
+        tke_ts = tke_series(u, v)
+        tke_avg = tke_spectrum(np.mean(u, axis=-1),
+                               np.mean(v, axis=-1))
+        return {'velocity_gradient': v_dist, 'vorticity': vort_dist,
+                'ramp_rate': rr_dist, 'tke_ts': tke_ts, 'tke_avg': tke_avg}
+
+    def get_height_stats(self, height):
+        """Get stats for high and low resolution wind fields
+
+        Parameters
+        ----------
+        height : int
+            Height in meters for requested U/V fields
+
+        Returns
+        -------
+        low_res : dict
+            Dictionary of stats for low resolution wind fields
+        high_res : dict
+            Dictionary of stats for high resolution wind fields
+        interp : dict
+            Dictionary of stats for spatiotemporally interpolated wind fields
+        """
+        u_hr, v_hr = self.get_uv_out(height)
+        uidx, vidx = self.feature_indices(height)
+        u_lr = self.source_handler.data[..., self.lr_t_slice, uidx]
+        v_lr = self.source_handler.data[..., self.lr_t_slice, vidx]
+        u_itp = st_interp(u_lr, self.s_enhance, self.t_enhance)
+        v_itp = st_interp(v_lr, self.s_enhance, self.t_enhance)
+
+        low_res = self.get_wind_stats(u_lr, v_lr)
+        high_res = self.get_wind_stats(u_hr, v_hr)
+        interp = self.get_wind_stats(u_itp, v_itp)
+        return low_res, high_res, interp
```

Comment:
Nice work, this is nice and clean and i can see why this is going to be super useful for all the wind actuals runs.

Reply:
Thanks!